crates/proto/src/rr/resource.rs

```diff
@@ -285,7 +333,22 @@ impl<'r> BinDecodable<'r> for Record {
                 decoder.read_u16()?.unverified(/*restricted to a min of 512 in for_opt*/),
             )
         } else {
-            DNSClass::read(decoder)?
+            #[cfg(not(feature = "mdns"))]
+            {
+                DNSClass::read(decoder)?
+            }
+
+            #[cfg(feature = "mdns")]
```

Comment:
I like this implementation better, doing it all in the Record feels more correct.

nice refactor.